### PR TITLE
Filter empty files when getting file list

### PR DESCRIPTION
**Pull request by hitchhicker**

### CODE REVIEW

File: autofaiss/readers/embeddings_iterators.py [hunk 1]
```diff
@@ -170,6 +179,24 @@ def get_matrix_reader(file_format: str, fs: fsspec.AbstractFileSystem, file_path
...
+    """Filter out all the empty files"""
+    if len(files) == 0:
+        return []
+    valid_files = []
+
+    def _is_empty_func(file_path):
+        return get_matrix_reader(file_format, file_system, file_path, None).is_empty(), file_path
+
+    with ThreadPool(min(16, len(files))) as pool:
+        for is_empty, file in pool.imap_unordered(_is_empty_func, files):
```
Comment (rom1504) [inline autofaiss/readers/embeddings_iterators.py:192]: imap_unordered is unordered, you're losing the order there

Comment (hitchhicker) [reply autofaiss/readers/embeddings_iterators.py:192]: Yes, I know. That's why I order them after the filter.

Comment (hitchhicker) [reply autofaiss/readers/embeddings_iterators.py:192]: Do you mean once in `read_total_nb_vectors_and_dim` and once in `_get_chunk_sizes` of distirbuted ?

Comment (rom1504) [reply autofaiss/readers/embeddings_iterators.py:192]: yes

Comment (rom1504) [reply autofaiss/readers/embeddings_iterators.py:192]: I don't mean that the code is duplicated, that's ok
I mean this is actually run twice and on hdfs it's not that fast to read thousand of files

Comment (rom1504) [reply autofaiss/readers/embeddings_iterators.py:192]: do we still need this function ?

Comment (hitchhicker) [reply autofaiss/readers/embeddings_iterators.py]: I removed it in the latest commit with some refactorings.

**Pull request merged**